lib/mquery.js

```diff
@@ -1426,12 +1426,17 @@ function _pushMap (opts, map) {
  * limit, skip, maxScan, batchSize, comment
  *
  * Sets these associated options.
+ * With mongodb>=3.2.6 if the value is a string doesn't accept it, converting type string numbers to numeric type in applicable cases
  *
  *     query.comment('feed query');
  */
 
 ;['limit', 'skip', 'maxScan', 'batchSize', 'comment'].forEach(function (method) {
   Query.prototype[method] = function (v) {
+    if (method !== 'comment') {
+      let value = Number(v);
+      v = (Number.isNaN(value)) ? v : value;
+    }
```

Comment:
I prefer strict validation over attempting to fix invalid input. If anything, let's assert the value is a number so the engineer fixes their code.